src/Context/Swoole/README.md

```diff
@@ -8,3 +8,91 @@
 This is a read-only subtree split of https://github.com/open-telemetry/opentelemetry-php-contrib.
 
 # OpenTelemetry Swoole context
+
+[简体中文](README-zh_CN.md) | [ENGLISH](README.md)
+
+Unlike the PHP-FPM runtime where a single process handles one request at a time, in Swoole, each process handles multiple http requests simultaneously. This library solves the context switching issue when using Opentelemetry in Swoole.
+
+## Requirement
+
+* php >= 8.0
+* swoole >= 4.5
+
+## Installation
+
+Install the package with composer:
+
+```bash
+composer require open-telemetry/context-swoole
+```
+
+Note: this library needs to be used in conjunction with OpenTelemetry, such as `open-telemetry/opentelemetry`.
+
+## Usage
+
+1.Register `TracerProvider` during the service startup:
+
+```php
+<?php
+
+use OpenTelemetry\API\Globals;
+use OpenTelemetry\API\Instrumentation\Configurator;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\ContextStorage;
+use OpenTelemetry\Contrib\Context\Swoole\SwooleContextStorage;
+use OpenTelemetry\Contrib\Otlp\OtlpHttpTransportFactory;
+use OpenTelemetry\Contrib\Otlp\SpanExporter;
+use OpenTelemetry\SDK\Common\Util\ShutdownHandler;
+use OpenTelemetry\SDK\Trace\SpanProcessor\SimpleSpanProcessor;
+use OpenTelemetry\SDK\Trace\TracerProvider;
+
+$transport = (new OtlpHttpTransportFactory())->create('http://collector:4318/v1/traces', 'application/json');
+$exporter  = new SpanExporter($transport);
+
+$spanProcessor = new SimpleSpanProcessor($exporter);
+$tracerProvider = new TracerProvider($spanProcessor);
+
+// Use Swoole context storage
+Context::setStorage(new SwooleContextStorage(new ContextStorage()));
+
+ShutdownHandler::register([$tracerProvider, 'shutdown']);
+
+Globals::registerInitializer(
+    fn (Configurator $configurator) => $configurator
+        ->withTracerProvider($tracerProvider)
+);
+```
+
+2.Execute the trace generation in the business (for example, in the `onRequest` event callback)
+
+```php
+<?php
+
+use OpenTelemetry\API\Globals;
+use OpenTelemetry\API\Trace\SpanKind;
+
+$tracer = Globals::tracerProvider()->getTracer('io.opentelemetry.contrib.swoole.php');
+
+$root = $span = $tracer->spanBuilder('root')->setSpanKind(SpanKind::KIND_SERVER)->startSpan();
+$scope = $span->activate();
+
+for ($i = 0; $i < 3; $i++) {
```

Comment:
could this example use swoole? eg an `onRequest` callback like the documentation mentions?

Reply:
Added, if there is anything else needed, please feel free to point it out.